use openpyxl for XLSX files; upgrade xlrd to v2

sheppard opened the pull request:
Closes #501

#### Why is this the best possible solution? Were any other approaches considered?

See discussion in #501.

#### What are the regression risks?

There should be no noticeable changes for most users of pyxform.  I started from a line-for-line translation of the xlrd integration and optimized some of the logic.  One potential change is that dictionary keys will not contain consecutive inner spaces, since they are now populated from the cleaned headers rather than a separate step.

#### Does this change require updates to documentation? If so, please file an issue [here](https://github.com/XLSForm/xlsform.github.io) and include the link below.

There is a new dependency, but this should be transparent to most users.

#### Before submitting this PR, please make sure you have:
- [X] included test cases for core behavior and edge cases in `tests`
- [X] run `nosetests` and verified all tests pass
- [X] run `black pyxform tests` to format code
- [X] verified that any code or assets from external sources are properly credited in comments

lindsay-stevens commented:
Hi @sheppard thanks for adding this PR! I have a few requests to consider please. If you don't expect to have the bandwidth to address these please let us know, and they could be done in a follow-up PR.

- Pin setup.py dependencies to latest versions. Pyxform users may have older versions which may not necessarily work: `xlrd==2.0.1` and `openpyxl==3.0.9`.
- A requirement for adding openpyxl should be to include defusedxml, which seems to involve:
  - add a setup.py dependency pinned to `defusedxml==0.7.1` so it is importable by openpyxl.
  - `os.environ["OPENPYXL_DEFUSEDXML"] = "True"` in xls2json_backends.py, as per: [openpyxl src](https://foss.heptapod.net/openpyxl/openpyxl/-/blob/5645879037fdf6f51292ec40edc31aa64e0ec1f6/openpyxl/xml/__init__.py) and [docs](https://openpyxl.readthedocs.io/en/stable/index.html#security).
- Add a test to test_xls2json_backends.py to explicitly exercise XLS (xls_to_dict) and XLSX (xlsx_to_dict) reading. This could be copy of [CsvReaderEquivalencyTest](https://github.com/XLSForm/pyxform/blob/66c0d5b9ad4a34bce47fe52a60e4366c19ee457c/tests/xls2json_tests.py#L210-L227) for XLS / XLSX instead of CSV / XLS.
- Move utils.py function-level imports to module level, since they are either internal imports or package-level dependencies. I'm not sure why there was a function-level import in sheet_to_csv.

sheppard commented:
I updated the setup.py dependencies, added equivalency tests, and moved the imports to the module level in utils.py.

It looks like openpyxl's defusedxml integration is active by default as long as:
 * defusedxml is installed, and
 * `OPENPYXL_DEFUSEDXML` is either `True` **or not set**.  (In spite of the name, `defusedxml_env_set()` returns `True` if the key does not exist in the dictionary.)

I confirmed this by adding a test to `assertTrue(openpyxl.DEFUSEDXML)` in test_xls2json_backends.py.

Setting `os.environ` would only be necessary to override the preferences of a user that had explicitly defined `OPENPYXL_DEFUSEDXML=False` in their environment.  In that case, the user would still be able to disable defusedxml by importing openpyxl before pyxform had a chance to update `os.environ`.

lindsay-stevens reviewed (approved):
Great! Thanks very much
